### PR TITLE
Feature/post slack

### PR DESCRIPTION
**Pull request by DerMeister**
feedback please :)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -87,3 +87,8 @@ ENV/
 
 # Rope project settings
 .ropeproject
+
+# pycharm
+.idea/
+log/
+api.log.json
```
Comment (pythonixpp) [inline .gitignore]: maybe just ignore all json files, or is there a case where u cannot ignore all? (Unit Test Resources etc.)

File: config.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+""" Configurations for the application."""
+from enum import Enum
+
+production = {
```
Comment (pythonixpp) [inline config.py]: May use the Overwritable configuration from the commonspy here. This makes the deployment easier.

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Flask
+flask-autodoc
+tornado
+requests
+pymongo
+flask-cors
```
Comment (pythonixpp) [inline requirements.txt]: do u realy need all those dependencies? I think u don't need flask-autodoc, tornado, flask-cors, pymongo

File: slacknotifier/flask_api.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+from slacknotifier.slack import slack
+
+
+def create_app():
+    app = Flask(__name__)
+    CORS(app)
+    app.config[AppConfig.cors_headers] = AppConfig.cors_headers_value
+    app.register_blueprint(slack, url_prefix=AppConfig.server_slack_prefix.value)
+
+    log_debug(AppConfig.server_root_url.value + AppConfig.server_slack_prefix.value + "/notify")
```
Comment (pythonixpp) [inline slacknotifier/flask_api.py]: May u use string formatting here like '%s%s/notify' % (AppConfig.server_root.value, AppConfig.server_slack_prefix.value)

File: tests/slack.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+from slacknotifier import slack
+
+
+class FullMockSlackConnector(slack.SlackConnector):
+    def push_notification(self, message):
+        pprint(message)
+
+    def request_top(self, group, n, req_date):
+        return [
```
Comment (pythonixpp) [inline tests/slack.py]: that is just awesome :-)

**Comment by pythonixpp** [comment]
Quite awesome code. I detected some small issues, but overall it looks good

**Comment by DerMeister** [comment]
Changed stuff, this time with a committed controller ...

**Comment by DerMeister** [comment]
@pythonixpp what should I use for the scheduled execution?
I found an easy to use scheduler under MIT license: https://github.com/dbader/schedule

**Pull request merged**